rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -239,6 +240,54 @@ public void savePegoutsWaitingForConfirmations() {
         }
     }
 
+    public Optional<SortedMap<Sha256Hash, List<Coin>>> getReleasesOutpointsValues() {
+        if (!activations.isActive(RSKIP305)) {
+            return Optional.empty();
+        }
+
+        if (releasesOutpointsValues != null) {
+            return Optional.of(releasesOutpointsValues);
+        }
+
+        releasesOutpointsValues = safeGetFromRepository(
+            RELEASES_OUTPOINTS_VALUES,
+            BridgeSerializationUtils::deserializeReleasesOutpointsValues
+        );
+
+        return Optional.ofNullable(releasesOutpointsValues)
+            .filter(map -> !map.isEmpty())
+            .map(Collections::unmodifiableSortedMap);
+    }
+
+    public void setReleaseOutpointsValues(Sha256Hash releaseTxHash, List<Coin> outpointsValues) {
+        if (releaseTxHash == null || outpointsValues == null) {
```

Comment:
Do we need the `if (!activations.isActive(RSKIP305))` check here? I know that the `save...` and `get...` methods already have it and that it might be enough. But this one could set `releasesOutpointsValues` to a map that will never be used.